realm/realm-library/src/main/java/io/realm/DynamicRealmObject.java

```diff
@@ -1365,6 +1422,138 @@ private void setModelDictionary(String fieldName, RealmDictionary<DynamicRealmOb
         }
     }
 
+    /**
+     * Sets the reference to a {@link RealmSet} on the given field.
+     * <p>
+     * This will copy all the elements in the set into Realm, but any further changes to the set
+     * will not be reflected in the Realm. Use {@link #getSet(String)} in order to get a reference to
+     * the managed set.
+     *
+     * @param fieldName field name.
+     * @param set       set of objects. Must either be primitive types or {@link DynamicRealmObject}s.
+     * @throws IllegalArgumentException if field name doesn't exist, it is not a set field, the objects in the
+     *                                  set doesn't match the expected type or any Realm object in the set
+     *                                  belongs to a different Realm.
+     */
+    public <E> void setSet(String fieldName, RealmSet<E> set) {
```

Comment:
Nitpicking big time: what about this instead?
```suggestion
    public <E> void setRealmSet(String fieldName, RealmSet<E> set) {
```
To me `setSet` seems a bit Inception-y.